src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseParticipationResource.java

```diff
@@ -248,4 +282,21 @@ public ModelAndView redirectGetParticipationRepositoryFiles(@PathVariable long p
         return new ModelAndView("forward:/api/repository/" + participation.getId() + "/files-content/" + commitId);
     }
 
+    /**
+     * GET /programming-exercise-participations/{participationId}/files-content-commit-details/{commitId} : Get the content of the files of a programming exercise participation.
+     * This method is specifically for the commit details view, where not only Instructors and Admins should have access to the files content as in
```

Comment:
this javadoc comment seems to contradict the logic of the method because it only enforces student

Reply:
I wrote it slightly better. It enforces at least student. The method above enforces at least instructor